core/src/main/java/org/fao/geonet/resources/CMISResources.java

```diff
@@ -0,0 +1,491 @@
+
+        for (CmisObject object : children) {
+            if (object instanceof Document) {
+                final Path curPath = getKeyPath(((DocumentImpl) object).getPaths().get(0));
+                try {
+                    if (iconFilter.accept(curPath)) {
+                        result.add(curPath);
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    @Nullable
+    @Override
+    public FileTime getLastModified(final Path resourcesDir, final ServletContext context,
+                                    final Path appPath, final String filename) throws IOException {
+        final Path file = locateResource(resourcesDir, context, appPath, filename);
+        final String key = getKey(file);
+        try {
+            final CmisObject object = CMISConfiguration.getClient().getObjectByPath(key);
+            if (object == null) {
+                return null;
+            } else {
+                return FileTime.from(object.getLastModificationDate().toInstant());
+            }
+        } catch (CmisObjectNotFoundException e) {
+            // Ignore not found error.
+        }
+        // key does not exist
+        return null;
+    }
+
+    @Override
+    public void deleteImageIfExists(final String image, final Path dir) {
+        Path icon = findImagePath(image, dir);
+        if (icon != null) {
+            CMISConfiguration.getClient().deleteByPath(getKey(icon));
+        }
+    }
+
+    private class CMISResourceHolder implements ResourceHolder {
+        private final String key;
+        private Path path = null;
+        private boolean writeOnClose = false;
+
+        private CMISResourceHolder(final String key, boolean writeOnClose) {
+            this.key = key;
+            this.writeOnClose = writeOnClose;
+        }
+
+        @Override
+        public Path getPath() {
+            if (path != null) {
+                return path;
+            }
+            final String[] splittedKey = key.split(CMISConfiguration.getFolderDelimiter());
+            try {
+                path = java.nio.file.Files.createTempFile("", splittedKey[splittedKey.length - 1]);
+                try {
+                    final CmisObject object = CMISConfiguration.getClient().getObjectByPath(key);
+                    if (object == null) {
+                        if (writeOnClose) {
+                            Files.delete(path);
+                        }
+                    } else {
+                        try (InputStream in = ((Document) object).getContentStream().getStream()) {
+                            java.nio.file.Files.copy(in, path,
+                                    StandardCopyOption.REPLACE_EXISTING);
+                        }
+                    }
+                } catch (CmisObjectNotFoundException e) {
+                    if (writeOnClose) {
```

Comment:
Should not be moved to a `finally` block?

Reply:
I don't think a try finally block will work here as it is only removing the path if the object does not exists otherwise it is loading it from the cmis repository.